sdk/eventhub/azure-eventhubs/api_review.py

```diff
@@ -0,0 +1,235 @@
+
+    async def send(
+        self,
+        event_data: Union[EventData, EventDataBatch, Iterable[EventData]],
+        *,
+        partition_key: Union[str, bytes] = None,
+        partition_id: str = None,
+        timeout: float = None) -> None:
+
+    async def create_batch(self, events: Iterable[EventData] = None, max_size: int = None) -> EventDataBatch:
+
+    async def get_properties(self) -> Dict[str, Any]:
+
+    async def get_partition_ids(self) -> List[str]:
+
+    async def get_partition_properties(self, partition: str) -> Dict[str, str]:
+
+    async def close(self) -> None:
+
+
+class EventHubConsumerClient(ClientBaseAsync):
+
+    eh_name  # type: str
+
+    def __init__(
+            self,
+            host: str,
+            event_hub_path: str,
+            credential: TokenCredential,
+            *
+            partition_manager: PartitionManager,
+            load_balancing_interval: int = 10,
+            logging_enable: bool = False,
+            http_proxy: dict = None,
+            auth_timeout: float = 60,
+            user_agent: str = None,
+            retry_total: int = 3,
+            transport_type=None
+            **kwargs
+    ) -> None:
+
+    async def __aenter__(self):
+
+    async def __aexit__(self, *args):
+
+    @classmethod
+    def from_connection_string(cls, conn_str: str, **kwargs) -> EventHubConsumerClient:
+
+    async def receive(
+            self,
+            on_event: Callable[PartitionContext, EventData],
+            consumer_group: str,
+            *,
+            partition_id: str = None,
+            owner_level: int = None,
+            prefetch: int = 300,
+            track_last_enqueued_event_properties: bool = False,
+            initial_event_position=None,
+            on_error=None,
+            on_partition_initialize=None,
+            on_partition_close=None
+    ) -> None:
+
+    async def get_properties(self) -> Dict[str, Any]:
+
+    async def get_partition_ids(self) -> List[str]:
+
+    async def get_partition_properties(self, partition: str) -> Dict[str, str]:
+
+    async def close(self) -> None:
+
+
+class PartitionContext(object):
+
+    fully_qualified_namespace  # type: str
+    partition_id  # type: str
+    eventhub_name  # type: str
+    consumer_group_name  # type: str
+    owner_id  # type: str
+
+    def __init__(
+        self,
+        fully_qualified_namespace: str,
+        eventhub_name: str,
+        consumer_group_name: str,
+        partition_id: str,
+        owner_id: str,
+        partition_manager: PartitionManager =None) -> None:
+
+    async def update_checkpoint(self, event: EventData):
+
+
+class PartitionManager(ABC):
+
+    async def list_ownership(
+        self,
+        fully_qualified_namespace: str,
+        eventhub_name: str,
+        consumer_group_name: str) -> Iterable[Dict[str, Any]]:
+
+    async def claim_ownership(self, ownership_list: Iterable[Dict[str, Any]]) -> Iterable[Dict[str, Any]]:
```

Comment:
Nitpick: I would have named the method something else given that it *attempts* to claim ownerships and returns what things it actually succeeded in claiming (i.e. it does a `try_claim_ownership`)